### PR TITLE
[Doc] Implrove `thread::Builder`'s doc.

### PR DESCRIPTION
**Pull request by gamazeps**
Part of #29378 .

- Explains *why* we would use the builder instead ofthe free function.
- Changes the parent-child explanation for a spawned-caller in `thread::Builder::spawn`
- Adds a link to `io::Result` in `thread::Builder`
- Corrects the return type doc in `thread::Builder::spawn`

r? @rust-lang/docs

### CODE REVIEW

File: src/libstd/thread/mod.rs [hunk 1]
```diff
@@ -180,8 +180,34 @@ pub use self::local::{LocalKey, LocalKeyState};
...
+///
+/// The two configurations available are:
+///
+/// - [`name`]: allows to give a name to the thread which is currently
+///   only used in `panic` messages.
+/// - [`stack_size`]: specifies the desired stack size. Note that this can
+///   be overriden by the OS.
+///
+/// If the [`stack_size`] field is not specified, the stack size
+/// will be the `RUST_MIN_STACK` environment variable.  If it is
```
Comment (GuillaumeGomez) [inline src/libstd/thread/mod.rs]: Extra whitespace before "If it is".

File: src/libstd/thread/mod.rs [hunk 2]
```diff
@@ -180,8 +180,34 @@ pub use self::local::{LocalKey, LocalKeyState};
...
+/// not specified either, a sensible default will be set (2MB as
```
Comment (GuillaumeGomez) [inline src/libstd/thread/mod.rs]: I'm not sure we should give the default value. Especially if it can change depending the platform/architecture.

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]

**Comment by gamazeps** [comment]
Done :)

File: src/libstd/thread/mod.rs [hunk 3]
```diff
@@ -180,8 +180,33 @@ pub use self::local::{LocalKey, LocalKeyState};
...
+/// will be the `RUST_MIN_STACK` environment variable. If it is
+/// not specified either, a sensible default will be set.
+///
+/// If the [`name`] field is not specified, the thread will not be named.
+///
+/// The [`spawn`] method will take ownership of the builder and create an
+/// [`io::Result`] to the thread handle with the given configuration.
+///
+/// The [`thread::spawn`] free function uses a `Builder` with default
+/// configuration and `unwrap`s its return value.
```
Comment (GuillaumeGomez) [inline src/libstd/thread/mod.rs]: Missing link to the unwrap method. :p

Comment (gamazeps) [inline src/libstd/thread/mod.rs]: Done :p

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**